bundles/org.eclipse.ui.workbench/eclipseui/org/eclipse/ui/internal/Workbench.java

```diff
@@ -2714,12 +2721,88 @@ public static Object setRestartArguments(String workspacePath) {
 		if (command_line == null) {
 			return IApplication.EXIT_OK;
 		}
+		Path selectedWorkspace = Path.of(workspacePath);
+		try {
+			String workspaceLock = getWorkspaceLockDetails(selectedWorkspace.toUri().toURL());
+			if (workspaceLock != null && !workspaceLock.isEmpty()) {
+				String lockMessage = NLS.bind(WorkbenchSWTMessages.IDEApplication_workspaceCannotLockMessage2,
+						workspacePath);
+				String wsLockedError = lockMessage + System.lineSeparator() + System.lineSeparator()
+						+ NLS.bind(WorkbenchSWTMessages.IDEApplication_workspaceLockMessage,
+						workspaceLock);
+
+				MessageDialog.openError(PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell(),
+						WorkbenchSWTMessages.IDEApplication_workspaceCannotLockTitle,
+						wsLockedError);
+				return null;
```

Comment:
This method is used also in `IDEApplication.checkInstanceLocation()` which uses `null` internally as "valid workspace is selected".

Looks like `null` breaks logic in `org.eclipse.ui.internal.ide.application.IDEApplication.checkInstanceLocation(Shell, Map)`, however I see the `null` was also added in be51d14c6f825ce5f072dcbf771449620e789b6a and javadoc was not updated too. See https://github.com/eclipse-platform/eclipse.platform.ui/pull/1307 + https://github.com/eclipse-platform/eclipse.platform.ui/pull/1622.

The use of `null` should have been documented & validated for  `IDEApplication.checkInstanceLocation()` usage in be51d14c6f825ce5f072dcbf771449620e789b6a, but latest now it is a must have.